node/container.go

```diff
@@ -133,16 +140,44 @@ func (a *HostAgent) StartContainer(cancel <-chan interface{}, svc *service.Servi
 	}
 	defer masterClient.Close()
 
+	evaluatedService, err := masterClient.GetEvaluatedService(partialSvc.ID, instanceID)
+	if err != nil {
+		logger.WithError(err).Error("Failed to get service")
+		return nil, nil, err
+	}
+	// Update the service with the complete image name
+	evaluatedService.ImageID = imageName
```

Comment:
We need to value of `imageName` to setup the container properly.  Things shuffled around some in the changeset 41a0386